src/pages/patientView/structuralVairnt/column/AnnotationColumnFormatter.tsx

```diff
@@ -0,0 +1,179 @@
+
+export default class AnnotationColumnFormatter {
+    public static getData(
+        structuralVariantData: StructuralVariant[] | undefined,
+        oncoKbCancerGenes?: RemoteData<CancerGene[] | Error | undefined>,
+        oncoKbData?: RemoteData<IOncoKbData | Error | undefined>,
+        usingPublicOncoKbInstance?: boolean,
+        uniqueSampleKeyToTumorType?: { [sampleId: string]: string },
+        studyIdToStudy?: { [studyId: string]: CancerStudy }
+    ) {
+        let value: IAnnotation;
+
+        if (structuralVariantData) {
+            let oncoKbIndicator: IndicatorQueryResp | undefined = undefined;
+            let oncoKbStatus: IAnnotation['oncoKbStatus'] = 'complete';
+
+            let oncoKbGeneExist = false;
+            let isOncoKbCancerGene = false;
+            if (oncoKbCancerGenes && !(oncoKbCancerGenes instanceof Error)) {
+                oncoKbGeneExist =
+                    _.find(
+                        oncoKbCancerGenes.result,
+                        (gene: CancerGene) =>
+                            gene.oncokbAnnotated &&
+                            (gene.entrezGeneId ===
+                                structuralVariantData[0].site1EntrezGeneId ||
+                                gene.entrezGeneId ===
+                                    structuralVariantData[0].site2EntrezGeneId)
+                    ) !== undefined;
+                isOncoKbCancerGene =
+                    _.find(
+                        oncoKbCancerGenes.result,
+                        (gene: CancerGene) =>
+                            gene.entrezGeneId ===
+                                structuralVariantData[0].site1EntrezGeneId ||
+                            gene.entrezGeneId ===
+                                structuralVariantData[0].site2EntrezGeneId
+                    ) !== undefined;
+            }
+
+            // oncoKbData may exist but it might be an instance of Error, in that case we flag the status as error
+            if (oncoKbData && oncoKbData.result instanceof Error) {
+                oncoKbStatus = 'error';
+            } else if (oncoKbGeneExist) {
+                // actually, oncoKbData.result shouldn't be an instance of Error in this case (we already check it above),
+                // but we need to check it again in order to avoid TS errors/warnings
+                if (
+                    oncoKbData &&
+                    oncoKbData.result &&
+                    !(oncoKbData.result instanceof Error) &&
+                    oncoKbData.status === 'complete'
+                ) {
+                    oncoKbIndicator = this.getIndicatorData(
+                        structuralVariantData,
+                        oncoKbData.result,
+                        uniqueSampleKeyToTumorType,
+                        studyIdToStudy
+                    );
+                }
+                oncoKbStatus = oncoKbData ? oncoKbData.status : 'pending';
```

Comment:
are we sure it's really pending if that thing doesn't exist?

Reply:
Looks like it is. Ref: https://github.com/cBioPortal/cbioportal-frontend/blob/master/src/pages/patientView/copyNumberAlterations/column/AnnotationColumnFormatter.tsx#L86